firebase-config/api.txt

```diff
@@ -65,6 +65,16 @@ package com.google.firebase.remoteconfig {
     method public int getLastFetchStatus();
   }
 
+  public class FirebaseRemoteConfigRealtimeUpdateFetchException extends com.google.firebase.remoteconfig.FirebaseRemoteConfigException {
+    ctor public FirebaseRemoteConfigRealtimeUpdateFetchException(@NonNull String);
+    ctor public FirebaseRemoteConfigRealtimeUpdateFetchException(@NonNull String, @Nullable Throwable);
+  }
+
+  public class FirebaseRemoteConfigRealtimeUpdateStreamException extends com.google.firebase.remoteconfig.FirebaseRemoteConfigException {
```

Comment:
What is the semantic difference between these 2 exceptions? Do we expect devs to understand this difference or is it part of the implementation detail?

Reply:
The difference is that the fetch exception is meant to indicate to the user that we got a new config update signal but we were unable to retrieve the latest config version that we were expecting. And the second one indicated that the stream was unable to open. We'll let users know about these exceptions so that they can expect them when writing their onError callback. We'll include these exceptions as part of our docs.

Reply:
Just for reference, do you have an example of different actions that would make sense to take depending on which exception is thrown?

Reply:
For the stream exception some things they could do is remove their listener, fetch again, or add a new listener to trigger another round of http stream retries.
And for the fetch exception users could try and fetch again or activate their default configs.

Reply:
And yeah, with the next PR that includes the retry methods we would do this for them already. This exception would be more of an indication to the user that Realtime won't be able to work at all due to the stream issue. This may enable them to revert to base RC as a fallback or some other mitigation where they are not reliant on realtime.

Reply:
Mainly so user's can fall back onto something else and know when to not rely on Realtime. When we retry we're only going to have limited attempts b/c we don't want to use up the devices bandwidth/battery making an unlimited amount of retries. Especially in scenarios when our endpoint is down or the device has no network connection and so it won't successfully connect for a while. Do you think we should just fail quietly and not pass anything back if we can't establish a stream?

Reply:
Updated. PTAL @vkryachko